live-build/misc/live-build-hooks/90-linux-migration-artifact.binary

```diff
@@ -0,0 +1,164 @@
+# depot/version.info           version and date information about the
+#                              upgrade image
+# depot/hashes                 the sha256sum of any upgrade artifacts
+#                              (see below)
+# depot/hashes.sig[.<release>] one cryptographic signature of depot/hashes
+#                              per Delphix release
+# depot/<upgrade artifacts>    generally cpio archives of OS and appliance
+#
+# As dx_upack.sh is the only piece of code that runs from the old
+# world, we treat it as a contract that lists the bare minimum
+# requirements for our artifact.
+#
+DEPOT_DIRECTORY=depot
+
+#
+# Delete any leftover files from previous runs.
+#
+rm -rf "$MIGRATION_ARTIFACT" "$DEPOT_DIRECTORY"
+
+#
+# Not all variants include the virtualization application (e.g.
+# internal-minimal) so don't bother generating a migration image
+# for those. /opt/delphix is always created as part of the
+# appliance. Thus, checking if it exists is a quick way for us to
+# distinguish between variants that have the appliance from
+# the ones that don't.
+#
+if [[ ! -d binary/opt/delphix ]]; then
+	echo "This artifact does not contain the appliance." \
+		" Skipping migration image..."
+	exit 0
+fi
+
+mkdir $DEPOT_DIRECTORY
+
+#
+# Generate archive for os root directory.
+#
+# Note: We temporarily change directory in the subshell (we move to the
+# "binary" directory) because find prints everything as a relative
+# path to the current directory and we don't want the generated archive
+# to have "binary" as its top-level directory.
+#
+(
+	cd binary
+	find ./* -print | cpio -oc
+) >$DEPOT_DIRECTORY/os-root.cpio
+
+#
+# Copy all migration scripts including dx_prepare which is needed by
+# dx_unpack.
+#
+cp migration-scripts/* $DEPOT_DIRECTORY
+
+#
+# There may be a version.info file in the current directory already.
+# That version.info is Linux-specific and does not necessarily have
+# all the fields that the illumos upgrade process requires. Thus, we
+# create our own special version info directly in the depot directory
+# with all the fields expected by dx_unpack.
+#
+{
+	#
+	# Note that both the following fields are bogus and supplied
+	# solely for making dx_unpack happy.
+	#
+	echo "DLPX_DATE=bogus"
+	echo "DLPX_OS_VERSION=Linux"
+
+	#
+	# DLPX_MIN_VERSION is specified for ensuring that we are
+	# migrating from the right version.
+	#
+	echo "DLPX_MIN_VERSION=5.3.0.0"
+
+	#
+	# DLPX_VERSION is set explicitly to match the version of the
+	# virtualization appliance, which is what upgrades in Linux
+	# do and therefore we try to stay consistent as this field
+	# really can be anything (e.g. a date, a point-version, a
+	# nickname, etc.)
+	#
+	# shellcheck disable=SC2016
+	echo "DLPX_VERSION=$(chroot binary dpkg-query \
+		-Wf '${Version}' delphix-virtualization)"
+} >$DEPOT_DIRECTORY/version.info
+
+#
+# Generate hashes file.
+#
+(
+	cd binary
+	find ./* -type f -print0 | xargs -0 sha256sum
```

Comment:
`find ./*` will ignore all dot files directly under `./`. You should use `find .` instead